### PR TITLE
E2E: Add the junit reporter

### PR DESCRIPTION
**Pull request by nunnatsa**
Remove the old test reporter (old junit + polarion) as they are not supported anymore. Fix the test command line to generate junit report.

Also, get rid of PanicOnError within tests and use gomaga matchers instead. Add retries when needed.

**Jira Ticket**:

```jira-ticket
None
```

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
no code or unit test changes
/override coverage/coveralls

File: tests/func-tests/utils.go [hunk 1]
```diff
@@ -48,16 +48,29 @@ func FlagParse() {
 
 func BeforeEach() {
 	virtClient, err := kubecli.GetKubevirtClient()
-	kvtutil.PanicOnError(err)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	Eventually(func() error {
+		return virtClient.RestClient().Delete().Namespace(kvtutil.NamespaceTestDefault).Resource("virtualmachines").Do(context.TODO()).Error()
+	}).WithTimeout(time.Minute).WithPolling(time.Second).Should(Succeed())
```
Comment (machadovilaca) [inline tests/func-tests/utils.go:55]: again I would maybe like to see this extracted to a method and have here something like this:

```
deleteAll("virtualmachines")
deleteAll("virtualmachineinstances")
deleteAll("persistentvolumeclaims")
```

**Comment by nunnatsa** [comment]
hco-e2e-operator-sdk-aws lane succeeded
/override ci/prow/hco-e2e-operator-sdk-azure
hco-e2e-operator-sdk-sno-azure lane succeeded
/override ci/prow/hco-e2e-operator-sdk-sno-aws
/test hco-e2e-upgrade-operator-sdk-sno-aws
/test hco-e2e-upgrade-operator-sdk-sno-azure
/test hco-e2e-upgrade-prev-operator-sdk-aws
/test hco-e2e-upgrade-prev-operator-sdk-azure

**Comment by nunnatsa** [comment]
hco-e2e-kv-smoke-gcp lane succeeded
/override ci/prow/hco-e2e-kv-smoke-azure

**Pull request merged**